dali/operators/reader/reader_op_test.cc

```diff
@@ -473,4 +475,275 @@ TEST(ReaderTestSimple, CheckNumSamples) {
   ASSERT_EQ(num_samples(3, 10), static_cast<Index>(std::ceil(10 * 1.0 / 3)));
 }
 
+class FileReaderTest : public DALITest {
+ public:
+  void SetUp() override {
+    constexpr int filecount = 16;
+
+    for (int i = 0; i < filecount; i++) {
+      std::string path = make_string("/tmp/dali_test_file_", to_string(i));
+      filepaths_.push_back(path);
+
+      // write a single byte to the file
+      std::ofstream f(path);
+      f << (uint8_t) i;
+    }
+  }
+
+  void TearDown() override {
+    for (const auto &path : filepaths_)
+      std::remove(path.data());
+  }
+
+  std::vector<std::string> filepaths_;
+
+  OpSpec MakeOpSpec() {
+      // Currently, only pad_last_batch=true is supported.
+      return OpSpec("FileReader")
+            .AddOutput("data_out", "cpu")
+            .AddOutput("labels", "cpu")
+            .AddArg("files", filepaths_)
+            .AddArg("checkpointing", true)
+            .AddArg("pad_last_batch", true);
+  }
```

Comment:
Maybe one more thing to check here - the `prefetch_queue_depth` could be different in or two cases? Although, the default is 1 which intuitevely is the most difficult case. :D